### ISSUE TITLE
Use GET and Base64 for GUIDs.

### ISSUE THREAD

**Issue by krhoyt** [krhoyt/Avocado#1]
Blog posts, Dev posts, and Medium posts, all offer a GUID option for getting individual items, but they do so through POST. The initial reason for this was that sometimes GUIDs take the form of a URL. Using a URL as part of a REST endpoint causes all variety of problems (extra slashes). 

GET /api/blog/post/:guid

The above request would error if the :guid was "http://kevinhoyt.com/some/post". So ...

POST /api/blog/post/guid

Was created where the actual GUID would be passed as POST body data (JSON). This works, but it always feels strange to issue a POST to GET an item. 

The proper way to address this problem would be to URL encode the GUID in the GET scenario, thus replacing slashes with other characters. However different platforms, and different browsers seem to have their own take on how to actually go about the encoding. As an alternative, and since the "encoding" work needs to happen anyways, use Base64 encoding.

In short, move from:

POST /api/blog/post/guid

To:

GET /api/blog/post/:guid

Where ":guid" is the Base64 representation of the GUID. Do this across Blog, Dev, and Medium, as they all have GUID fields.

Node.js:

    // Encode
    let data = 'kevinhoyt.com';
    let buff = new Buffer( data );
    let base64data = buff.toString( 'base64' );

    // Decode
    let data = 'c3RhY2thYnVzZS5jb20=';
    let buff = new Buffer( data, 'base64' );  
    let text = buff.toString( 'utf-8' );  

Python:

    # Encode
    data = "kevinhoyt.com"
    base64.b64encode( s.encode( 'utf-8' ) )

    # Decode
    data = 'c3RhY2thYnVzZS5jb20=';
    base64.b64decode( data ).decode( 'utf-8' )

**Comment by krhoyt** [comment]
Implemented base64 encoding/decoding on both the aggregation side (Python) and API endpoints (Node.js). Across API surface, wherever "GUID" is referenced, it is expected to be Base64, whether the value is a URL or other (consistency).
